### ISSUE TITLE
Sorting of "use" includes forward slash

### ISSUE THREAD

**Issue by thinkingmedia** [cakephp/cakephp-codesniffer#175]
phpcs was reporting that the following was in the wrong order.

```
use Cake\Controller\Component\CookieComponent;
use Cake\Controller\ComponentRegistry;
```

phpcs was reporting that the `\` character comes after `R`.

Is it possible to not include the slashes as part of the sort order. So that the following is valid.

```
use Cake\Controller\ComponentRegistry;
use Cake\Controller\Component\CookieComponent;
```

This is the default for how phpstorm sorts use statements, and I find it more readable.

**Comment by dereuromark** [comment]
I remember that I once said the same thing in a PR :)

**Comment by dereuromark** [comment]
Correct me if I am wrong, but [mine](https://github.com/php-fig-rectified/psr2r-sniffer/blob/master/PSR2R/Sniffs/Namespaces/UseInAlphabeticalOrderSniff.php) should behave correctly. Maybe, if confirmed, that can be ported to this repo?

**Comment by markstory** [comment]
@dereuromark You sniff also reports the classes as being in the 'wrong' order as well.

**Comment by markstory** [comment]
> phpcs was reporting that the \ character comes after R.

This is probably because `\` does come after `R` in the ascii table which is generally how string values are sorted. Changing the sort order will prevent tools/editors doing simple string sorting on `use` classes.

**Comment by ADmad** [comment]
Is the sniff which reports this error in our repo or coming from the PSR2 ruleset from phpcs's repo? If it's the latter than we surely shouldn't change it.

**Comment by markstory** [comment]
@ADmad It is in our repo. The PSR2 sniffs don't make any assertions around use statement ordering.

**Comment by dereuromark** [comment]
The same issue came up in https://github.com/spryker/code-sniffer/issues/74 - which also uses our sniffer basically.
I think we should fix it to be consistent with the IDEs and in general the expectation of ordering people have.

> Changing the sort order will prevent tools/editors doing simple string sorting on use classes.

@markstory It is actually the other way around, those tools currently seem to reorder it by their expectation, and our sorting right now is the opposite.

**Comment by markstory** [comment]
@dereuromark As a counter example, vim uses the current ordering :smile:. I'm fine with the ordering changing if more tools do it the 'other' way though.

**Comment by dereuromark** [comment]
My colleague found an issue [fix](https://github.com/cakephp/cakephp-codesniffer/issues/175)
If enough positive feedback on this subject is given (agree to change), we can start a PR for the CakePHP sniffer, as well.

**Comment by josbeir** [comment]
Also experiencing sort issues in sublime and vscode (haven't tried anything else). Bit frustrating as i mostly use a "sort use on save" function.

Alphabetically sorted by above mentioned editors:
```php
use Cake\Controller\Component;
use Cake\Controller\Component\SecurityComponent;
use Cake\Controller\ComponentRegistry;
use Cake\Event\Event;
```

What the sniff expects:
```php
use Cake\Controller\Component;
use Cake\Controller\ComponentRegistry;
use Cake\Controller\Component\SecurityComponent;
use Cake\Event\Event;
```

**Comment by ADmad** [comment]
We can/should change this in next major release. The sort order our sniff expects is at odds with most other tools.

**Comment by josbeir** [comment]
In the meantime... and if you are using vscode, i added a 'natural sorting' option to the namespace resolver extension => https://github.com/MehediDracula/PHP-Namespace-Resolver/pull/29 which does the trick =-)

**Comment by ADmad** [comment]
@josbeir Sweet

**Comment by ADmad** [comment]
Done for `next` branch.
